runners/core-java/src/main/java/org/apache/beam/runners/core/ReduceFnRunner.java

```diff
@@ -919,8 +920,9 @@ private boolean needToEmit(boolean isEmpty, boolean isFinished, PaneInfo.Timing
       // The pane has elements.
       return true;
     }
-    if (timing == Timing.ON_TIME) {
-      // This is the unique ON_TIME pane.
+    if (timing == Timing.ON_TIME
+        && windowingStrategy.getOnTimeBehavior() == Window.OnTimeBehavior.FIRE_ALWAYS) {
```

Comment:
I think this is correct. I think `OnTimerBehavior.FIRE_ALWAYS` should be the same behavior as before this change. Actually, it is required to be the same for backwards compatibility, anyhow.